src/Test/TestDamageStrategy.java

```diff
@@ -0,0 +1,138 @@
+    // this order is needed because that's how the unit and weaponmodel enums are currently set up.
+    unitModels = new ArrayList<UnitModel>(19);
+    unitModels.add(new InfantryModel());
+    unitModels.add(new MechModel());
+    unitModels.add(new ReconModel());
+    unitModels.add(new TankModel());
+    unitModels.add(new MDTankModel());
+    unitModels.add(new NeotankModel());
+    unitModels.add(new APCModel());
+    unitModels.add(new ArtilleryModel());
+    unitModels.add(new RocketsModel());
+    unitModels.add(new AntiAirModel());
+    unitModels.add(new MobileSAMModel());
+
+    unitModels.add(new FighterModel());
+    unitModels.add(new BomberModel());
+    unitModels.add(new BCopterModel());
+    unitModels.add(new TCopterModel());
+
+    unitModels.add(new BattleshipModel());
+    unitModels.add(new CruiserModel());
+    unitModels.add(new LanderModel());
+    unitModels.add(new SubModel());
+
+    // We build an array of weapons by taking each unit's weapons in turn.
+    weaponModels = new ArrayList<WeaponModel>();
+    for( UnitModel unit : unitModels )
+    {
+      if( null != unit.weaponModels )
+      {
+        for( WeaponModel weapon : unit.weaponModels )
+        {
+          weaponModels.add(weapon);
+        }
+      }
+    }
+
+  }
+
+  @Override
+  public boolean runTest()
+  {
+    setupTest();
+
+    boolean testPassed = true;
+    testPassed &= validate(true, "  Everything you know is wrong.");
+//    System.out.println(getDeltaString(new StandardDamage()));
+    return testPassed;
+  }
+
+  /**
+   * The values in the output array denote how much extra damage each interaction has with the new system compared to the standard. Output can be negative.
+   * @param other: which damage strategy to compare
+   * @return an array comprising the difference between the two damage strategy results
+   */
+  public int[][] getDeltas(DamageStrategy other)
+  {
+    int[][] deltas = new int[numWeaponModels][numUnitModels];
+    for( int i = 0; i < numWeaponModels; i++ )
+    {
+      WeaponModel weapon = weaponModels.get(i);
+      for( int j = 0; j < numUnitModels; j++ )
+      {
+        UnitModel target = unitModels.get(j);
+        deltas[i][j] = (int) (other.getDamage(weapon, target) - standardDamage.getDamage(weapon, target));
+      }
+    }
+    return deltas;
+  }
+
+  /**
+   * Uses getDeltas() to get the data it needs, then formats it decently for human consumption
+   * @return a string with the difference between the damage tables
+   */
+  public String getDeltaString(DamageStrategy other)
+  {
+    String out = "Defenders: ";
+    int[][] deltas = getDeltas(other);
+    for( UnitEnum type : UnitModel.UnitEnum.values() )
+    {
+      out += type.toString() + ", ";
+    }
+    out += "\n\n";
+    for( int i = 0; i < numWeaponModels; i++ )
+    {
+      out += WeaponModel.WeaponType.values()[i].toString() + ": ";
+      for( int j = 0; j < numUnitModels; j++ )
+      {
+        out += deltas[i][j] + ", ";
```

Comment:
The "print everything in a giant unformatted grid" approach is not the most readable - probably fine for now, but I might want to play with options at some point.
Also, just to make sure I'm reading this correctly - if I `getDelstaString` on `DSDamage`, it looks like `TANKCANNON`, `MD_TANKCANNON`, and `NEOCANNON` have negative values in the first two slots, which seems to imply that in either standard or DS damage strategies, cannons can shoot infantry?

Reply:
The AW2 damage table from Wars World News (made by Xenesis, the guy who does all the AW hacking) has damage numbers for tank cannons VS infantry.
We already had tanks shooting infantry with those numbers, I just fixed them to use machine guns.
Damage chart: https://www.warsworldnews.com/wp/aw2/game-aw2/damage-chart/

Reply:
FWIW, I'd really like to switch to AW3's damage chart for the time being, if we don't come up with a real damage system.

Reply:
What is better about AW3's chart?

Reply:
Cruisers, mainly. They are effective anti-air (instead of being worse AAs) and take significantly less damage from air units (they take the most damage from Bcopters in AW2 of any ship, I believe.)
They can also engage other ships with their anti-sub torpedoes, which is good since that means it can attack the Ship chassis generically, instead of having to special-case somehow.